### PR TITLE
Attitude servo behavior test

### PR DESCRIPTION
**Pull request by patapufate**
# Description
This PR Introduce the attitude servo behavior test. The test:

- Load the mission attitude_servo_mission_test.xml
- Execute the mission
- Check if the behavior publishes the goal
- Simulate PoseEstimator data to finish the behavior
- Test if the mission is SUCCESS

Also, it fixes a bug in the method that checks if the behavior has ended

## Type of change

Please mark options that are relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

# Checklist
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] The code builds clean without any errors or warnings
- [x] Unit tests are passing
- [x] Linter tests are passing

# How To Test
**Integration test** 
```sh
rostest mission_control test_mission_control_behaviors.test
```

### CODE REVIEW

File: mission_control/test/test_attitude_servo_behavior.py [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+        -   Check if the behavior publishes the goal
+        -   Simulate PoseEstimator data to finish the behavior
+        -   Test if the mission is SUCCESS
+    """
+
+    @classmethod
+    def setUpClass(cls):
+        rospy.init_node('test_attitude_servo_behavioral')
+
+    def setUp(self):
```
Comment (hidmic) [inline mission_control/test/test_attitude_servo_behavior.py:75]: @patapufate meta: could we start reusing test fixtures? Or at least the fraction these share. I see lots of copy and paste here and there.

**Review by hidmic — COMMENTED** [review]
Overall LGTM, except for one comment.

File: mission_control/test/test_attitude_servo_behavior.py (hunk 1 above)
Comment (patapufate) [reply mission_control/test/test_attitude_servo_behavior.py:75]: In these behavior test I really want to keep separated.

File: mission_control/src/behaviors/attitude_servo.cpp [hunk 2]
```diff
@@ -88,14 +90,19 @@ BT::NodeStatus AttitudeServoBehavior::behaviorRunningProcess()
...
+    behaviorStartTime_ = ros::Time::now();
     goalHasBeenPublished_ = true;
   }
   else
   {
     ros::Duration delta_t = ros::Time::now() - behaviorStartTime_;
     if (delta_t.toSec() > timeOut_) setStatus(BT::NodeStatus::FAILURE);
+    else
+    {
+      if (behaviorComplete_) setStatus(BT::NodeStatus::SUCCESS);
```
Comment (hidmic) [inline mission_control/src/behaviors/attitude_servo.cpp:102]: @patapufate consider checking for completion before checking for timeout.

Comment (hidmic) [reply mission_control/src/behaviors/attitude_servo.cpp:102]: Also, meta: stateful actions like these won't allow mission re-execution.

File: mission_control/test/test_attitude_servo_behavior.py [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+        self.attitude_servo_goal = AttitudeServo()
+        self.mission = MissionInterface()
+
+        # Subscribers
+        self.attitude_servo_msg = rospy.Subscriber('/mngr/attitude_servo',
+                                                   AttitudeServo, self.callback_publish_attitude_servo_goal)
```
Comment (hidmic) [inline mission_control/test/test_attitude_servo_behavior.py:70]: @patapufate nit: 
```suggestion
        self.attitude_servo_msg = rospy.Subscriber(
            '/mngr/attitude_servo',
            AttitudeServo, 
            self.callback_publish_attitude_servo_goal)
```
Same elsewhere.

File: mission_control/test/test_attitude_servo_behavior.py [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+
+        self.simulated_pose_estimator_pub = rospy.Publisher('/pose/corrected_data',
+                                                            CorrectedData, queue_size=1)
+
+    def callback_publish_attitude_servo_goal(self, msg):
```
Comment (hidmic) [inline mission_control/test/test_attitude_servo_behavior.py:75]: @patapufate nit: 
```suggestion
    def attitude_servo_goal_callback(self, msg):
```
? It reads better.

File: mission_control/test/test_attitude_servo_behavior.py [hunk 5]
```diff
@@ -0,0 +1,119 @@
...
+        self.attitude_servo_goal = msg
+        rospy.loginfo(msg)
+
+    def test_mission_with_attitude_servo_behavioral(self):
```
Comment (hidmic) [inline mission_control/test/test_attitude_servo_behavior.py:79]: @patapufate nit:
```suggestion
    def test_mission_with_attitude_servo_behavior(self):
```

File: mission_control/test/test_attitude_servo_behavior.py [hunk 6]
```diff
@@ -0,0 +1,119 @@
...
+        self.mission.load_mission("attitude_servo_mission_test.xml")
+        self.mission.execute_mission()
+        rospy.loginfo("Ejecutando")
```
Comment (hidmic) [inline mission_control/test/test_attitude_servo_behavior.py:82]: @patapufate nit: wrong language :sweat_smile:

File: mission_control/test/test_attitude_servo_behavior.py [hunk 7]
```diff
@@ -0,0 +1,119 @@
...
+        yaw = self.mission.get_behavior_parameter('yaw')
+        speed_knots = self.mission.get_behavior_parameter('speed_knots')
+
+        # Check if the behavior publishes the goal
+        def attitude_servo_goals_are_set():
+            return (self.attitude_servo_goal.roll == roll and
+                    self.attitude_servo_goal.pitch == pitch and
+                    self.attitude_servo_goal.yaw == yaw and
+                    self.attitude_servo_goal.speed_knots == speed_knots and
+                    self.attitude_servo_goal.ena_mask == 15)
```
Comment (hidmic) [inline mission_control/test/test_attitude_servo_behavior.py:96]: @patapufate where does this magic number come from?

File: mission_control/src/behaviors/attitude_servo.cpp (hunk 2 above)
Comment (patapufate) [reply mission_control/src/behaviors/attitude_servo.cpp:102]: Yes. Good catch. That's why I put behaviorStartTime_ when we publish the goal.

File: mission_control/test/test_attitude_servo_behavior.py (hunk 3 above)
Comment (patapufate) [reply mission_control/test/test_attitude_servo_behavior.py:70]: mmmm. This is strange roslint pep8 doesn't show error.
Fixed

File: mission_control/test/test_attitude_servo_behavior.py (hunk 4 above)
Comment (patapufate) [reply mission_control/test/test_attitude_servo_behavior.py:75]: Fixed.

File: mission_control/test/test_attitude_servo_behavior.py (hunk 5 above)
Comment (patapufate) [reply mission_control/test/test_attitude_servo_behavior.py:79]: Fixed

File: mission_control/test/test_attitude_servo_behavior.py (hunk 6 above)
Comment (patapufate) [reply mission_control/test/test_attitude_servo_behavior.py:82]: Fixed.

File: mission_control/test/test_attitude_servo_behavior.py (hunk 7 above)
Comment (patapufate) [reply mission_control/test/test_attitude_servo_behavior.py:96]: Changed to be calculated from mission parameters

**Pull request merged**